### PR TITLE
[lookup][runtime] making first column of runtime tables customizable

### PR DESCRIPTION
**Pull request by mimoo**
currently the first column of all runtime tables is an indexed column (0, 1, 2, ...)
this PR allows the first column to be determined arbitrarily at setup

### CODE REVIEW

File: kimchi/src/circuits/lookup/runtime_tables.rs [hunk 1]
```diff
@@ -10,16 +10,65 @@ use ark_ff::Field;
...
 }
 
+/// At setup time, a runtime table can be configured to be an indexed table,
+/// or a custom table.
+/// An index table has a counter starting from 0 in its first column,
+/// whereas a custom table has a given list of field elements.
+pub enum RuntimeTableSetup<F> {
+    /// An indexed runtime table has a counter, starting from zero, as first column.
+    Indexed(RuntimeTableConfiguration),
+    /// Unliked an indexed runtime table, the first column is fixed at setup time.
```
Comment (mrmr1993) [inline kimchi/src/circuits/lookup/runtime_tables.rs:30]: Nit: `Unlike`

Comment (mrmr1993) [reply kimchi/src/circuits/lookup/runtime_tables.rs:30]: Nit: both are fixed at setup time, it's just that the first column isn't necessarily sequential here.

**Review by mrmr1993 — APPROVED** [review]
A couple of nits, but LGTM

**Comment by mimoo** [comment]
wow you're fast o.o I forced push thinking I had some time, basically replaced the names:

* RuntimeTableConfiguration -> RuntimeTableSpec
* RuntimeTableSetup -> RuntimeTableCfg

**Comment by mimoo** [comment]
(note that this relies on https://github.com/o1-labs/proof-systems/pull/574)

**Review by mrmr1993 — APPROVED** [review]

**Pull request merged**